src/mapgen.cpp

```diff
@@ -3571,17 +3571,53 @@ class jmapgen_nested : public jmapgen_piece
                     return true;
                 }
         };
+        // TO DO: Remove this and replace it with a smarter logical syntax for neighbor_join_check and neighbor_flag_check
+        class neighbor_flag_any_check
+        {
+            private:
+                std::unordered_map<direction, cata::flat_set<oter_flags>> neighbors;
+            public:
+                explicit neighbor_flag_any_check( const JsonObject &jsi ) {
+                    for( direction dir : all_enum_values<direction>() ) {
+                        cata::flat_set<oter_flags> dir_neighbours;
```

Comment:
Nitpick: use consistent spelling of neighbo(u)rs.

Reply:
The bane of trying to abide by Americanisms when I'm not American